python/ray/data/datasource/parquet_datasource.py

```diff
@@ -246,6 +251,36 @@ def get_read_tasks(self, parallelism: int) -> List[ReadTask]:
 
         return read_tasks
 
+    def _estimate_decompression_multiplier(self) -> float:
+        """Return an estimate of the decompression multipler.
+
+        To avoid OOMs, it is safer to return an over-estimate than an underestimate.
+        """
+        # Sample a few rows from the row group to estimate the ratio.
+        # TODO(ekl/cheng) take into account column pruning.
+        num_samples = min(
+            PARQUET_COMPRESSION_RATIO_ESTIMATE_NUM_SAMPLES, len(self._pq_ds.pieces)
+        )
+        row_group_samples = [
+            p.subset(row_group_ids=[0]) for p in self._pq_ds.pieces[:num_samples]
```

Comment:
Sounds good, though it would be nice if the sampling was deterministic (overall) (so you don't get different behavior each time you read the dataset. Maybe we can read every kth row from every nth file or something like that.